maintenance/GenerateMissingIdentifiers.php

```diff
@@ -20,7 +21,22 @@ public function __construct() {
 	}
 
 	public function execute(): void {
-		$this->output( 'TODO' );
+		$pageIds = PersistentPageIdentifiersExtension::getInstance()->getPageIdsRepo()
+			->getPageIdsOfPagesWithoutPersistentIds();
+		$pageIdsCount = count( $pageIds );
+
+		foreach ( $pageIds as $pageId ) {
+			$this->generatePersistentIdForPage( intval( $pageId ) );
+		}
+
+		$this->output( "Created $pageIdsCount persistent IDs\n" );
+	}
+
+	private function generatePersistentIdForPage( int $pageId ): void {
```

Comment:
This name suggests no side effect

Reply:
`generateAndSavePeristentIdForPage`?

Reply:
That's better. It does highlight the function could be split, but whatever